integration/spark-common/src/main/scala/org/apache/carbondata/spark/CarbonOption.scala

```diff
@@ -31,6 +31,10 @@ class CarbonOption(options: Map[String, String]) {
 
   def tablePath: String = s"$dbName/$tableName"
 
+  def allowExisting: String = options.getOrElse("allowExisting", "true")
+
+  def tblProperties: String = options.getOrElse("tblproperties", "")
```

Comment:
it can be an option. in one way, it can  enhance dataFrame API, for example, .option("tblproperties", "'dictionary_incdlue='id''), so that user can decide for make dict or not.  in another way, it can support for using tableproperties when  CreateAsSelect .